src/app/_components/SearchInput.tsx

```diff
@@ -4,23 +4,27 @@ import { MagnifyingGlass } from '@phosphor-icons/react/dist/ssr'
 
 import { Icon } from '@/components/Icon'
 
+import { SearchProps } from './Search'
+
 type SearchInputProps = {
+  id: SearchProps['id']
   searchQuery: string
   onSearchChange: (event: ChangeEvent<HTMLInputElement>) => void
 }
```

Comment:
How about doing it this way?

```suggestion
type SearchInputProps = {
  searchQuery: string
  onSearchChange: (event: ChangeEvent<HTMLInputElement>) => void
} & React.ComponentProps<'input'>
```


Since `id` is one of the native attributes of the `input` element, we can "import" it with a union type instead of typing it manually.

What do you think?

Reply:
Oh that's cool, didn't know about that.

One problem might be that typescript then doesn't show errors if we don't include `id` as a prop in `SearchInput`.

### with React.ComponentProps<'input'>
<img width="932" alt="Screenshot 2024-06-17 at 12 02 20" src="https://github.com/FilecoinFoundationWeb/filecoin-foundation/assets/50910606/1965b457-8f54-438f-b55a-89bd02cf9d29">


### with  id: SearchProps['id']

<img width="932" alt="Screenshot 2024-06-17 at 12 00 05" src="https://github.com/FilecoinFoundationWeb/filecoin-foundation/assets/50910606/080ce41e-6e92-4ba8-8308-12933f55856a">

Reply:
I see. How about something like this?

```ts
type SearchInputProps = {
  id: React.ComponentProps<'input'>['id']
  searchQuery: string
  onSearchChange: (event: ChangeEvent<HTMLInputElement>) => void
}
```

In the future, we might want to use `SearchInput` separately from `Search` so that the type declaration seems more accurate.

Reply:
gotcha, that works :)